Makefile

```diff
@@ -36,7 +36,21 @@ STRESSFLAGS  :=
 DUPLFLAGS    := -t 100
 BUILDMODE    := install
 
-# Note: We pass `-v` go `go build` and `go test -i` so that warnings
+# If we are in the default directory (GOPATH/src/github.com/cockroachdb/cockroach), we can set
+# GOPATH automatically. Otherwise (e.g. vendoring), leave it as is.
+GOPATH_CANDIDATE := $(realpath ../../../..)
+
+ifneq ($(wildcard $(GOPATH_CANDIDATE)/src/.),)
+ifneq ($(wildcard $(GOPATH_CANDIDATE)/pkg/.),)
```

Comment:
Only `src` is guaranteed to exist; the others may not be created until after you've built something. (I'm not sure about the order of execution here but `glock` will create `bin` when it runs. I don't think it will make `pkg`, though)